### PR TITLE
Develop updates to Master

### PR DESCRIPTION
**Pull request by BadBrahmin**
Contains -

- Develop,
- Admin-Dashboard,
- MapUI,
- Header,
- Login-Updates

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
-            <Route path="/admin" component={AdminLogin} />
-            <Route exact path="/admindashboard" component={AdminDashboard} />
-            <Route path="/allcontent" component={AllContents} />
-            <Route path="/addcontent" component={AddContent} />
-            <Route path="/students" component={Students} />
+            <Route path="/register" component={RegisterUser} />
+            <Route path="/login" component={LoginUser} />
+            <Route path="/admin/login" component={AdminLogin} />
+            <Route exact path="/admin/dashboard" component={AdminDashboard} />
+            <Route path="/content/list" component={ContentList} />
```
Comment (prank7) [inline client/components/App.jsx]: Everything under admin dashboard goes under the url of `/admin`. So `/admin/content/new`, `/admin/pending-approvals` etc.

File: client/components/Header.js [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
+import { Link } from "react-router-dom"
 
 class Header extends React.Component {
+
+  handleLogout = () => {
+    localStorage.clear()
+    this.redirect()
+  }
+
+  redirect = () => {
```
Comment (prank7) [inline client/components/Header.js]: If `redirect` is being called only once then it shouldn't be an extra function just write it in `handleLogout` function itself.

File: client/components/adminDashboard/PendingApprovals.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class PendingApprovals extends Component {
+  constructor() {
+    super();
+    this.state = {
+      pendingStudentList: null
+    };
+  }
+
+  componentDidMount() {
+    fetch("/api/v1/admin/pending", {
```
Comment (prank7) [inline client/components/adminDashboard/PendingApprovals.js]: change the url to something more explicit. `/admin/students/pending`.

Comment (prank7) [reply client/components/adminDashboard/PendingApprovals.js]: Also, it should be part of redux action-creators

File: routes/userRouter.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+router.post("/register", users.registerUser);
+
+router.post("/login", users.loginUser);
+
+// User Status Route
+
+//TODO
+//Relocate this line
+router.get("/all", auth.verifyToken, users.userStatus);
```
Comment (prank7) [inline routes/userRouter.js]: `/all` needs to be changed.

File: utils/auth.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+var jwt = require('jsonwebtoken');
+
+// Exporting The Module Of Jwt Token
+ function generateToken (payload) {
+	return jwt.sign(payload, "abcdef");
+};
+
+// Verify The  Token
+function verifyToken (req, res, next) {
+	var Token = req.headers.authorization || '';
```
Comment (prank7) [inline utils/auth.js]: `Token` has to be lowercase `token`.

**Review by prank7 — CHANGES_REQUESTED** [review]

File: client/components/Header.js (hunk 2 above)
Comment (BadBrahmin) [reply client/components/Header.js]: @AmanSharma4419 Can you please fix this?

Comment (BadBrahmin) [reply client/components/Header.js]: @akhildhiman Can you please fix this?

File: routes/userRouter.js (hunk 4 above)
Comment (BadBrahmin) [reply routes/userRouter.js]: Changed to "/status/all"

File: client/components/Header.js (hunk 2 above)
Comment (akhildhiman) [reply client/components/Header.js]: @BadBrahmin I've updated it. Please check.

File: client/components/App.jsx (hunk 1 above)
Comment (BadBrahmin) [reply client/components/App.jsx]: done

File: client/components/adminDashboard/PendingApprovals.js (hunk 3 above)
Comment (BadBrahmin) [reply client/components/adminDashboard/PendingApprovals.js]: Route changed. 

Pending: Implementation in redux.

**Pull request merged**